### PR TITLE
Feature/wid 209 - angular update

### PR DESCRIPTION
**Pull request by brampauwelyn**
# Migrated from angular v4.4.7 to v8.1.2 :tada:

### Changed
- updated package.json to upgrade angular packages and all it's dependencies.
- fixed css issues after updating bootstrap  to  and ng-bootstrap to v5.
- fixed all vulnerabilities :fire:

### Steps to test / get changes locally
```
$ git fetch --all
$ git checkout feature/WID-209
$ rm -rf node_modules
$ rm package-lock.json
$ sudo npm install
$ ng serve 
```

### CODE REVIEW

File: src/app/core/route/resolver/project-resolver.service.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { Injectable } from '@angular/core';
+import {of as observableOf,  Observable } from 'rxjs';
```
Comment (reinoutbossuyt) [inline src/app/core/route/resolver/project-resolver.service.ts]: Shouldn't you place a space before it?

File: src/app/core/route/resolver/project-resolver.service.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
+import {catchError, map} from 'rxjs/operators';
```
Comment (reinoutbossuyt) [inline src/app/core/route/resolver/project-resolver.service.ts]: Shouldn't you place a space before it?

File: src/app/core/project/services/project.service.ts [hunk 3]
```diff
@@ -30,15 +29,15 @@ export class ProjectService {
...
-    return this.http.get(environment.apiUrl + 'project/' + id)
-      .map(project => new Project(project))
-      .do(project => {
+    return this.http.get(environment.apiUrl + 'project/' + id).pipe(
+      map(project => new Project(project)),
+      tap(project => {
         // Cache the response
         this.cache.put('project', [id], project);
-      });
+      }),);
```
Comment (reinoutbossuyt) [inline src/app/core/project/services/project.service.ts]: Komma

File: src/app/core/project/services/project.service.ts [hunk 4]
```diff
@@ -1,9 +1,8 @@
 import { Injectable } from '@angular/core';
+import {of as observableOf, Observable } from 'rxjs';
```
Comment (reinoutbossuyt) [inline src/app/core/project/services/project.service.ts]: space

File: src/app/core/project/services/project.service.ts [hunk 5]
```diff
@@ -1,9 +1,8 @@
...
+import {tap, map} from 'rxjs/operators';
```
Comment (reinoutbossuyt) [inline src/app/core/project/services/project.service.ts]: space

File: src/app/core/user/services/user.service.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import { Injectable } from '@angular/core';
+import {of as observableOf,  Observable } from 'rxjs';
+import {tap, map, catchError} from 'rxjs/operators';
 import { environment } from '../../../../environments/environment';
 import { HttpClient } from '@angular/common/http';
-import { Observable } from 'rxjs/Observable';
-import 'rxjs/add/operator/do';
-import 'rxjs/add/operator/map';
+
```
Comment (reinoutbossuyt) [inline src/app/core/user/services/user.service.ts]: Why adding two linebreaks?

File: src/index.html [hunk 7]
```diff
@@ -7,7 +7,14 @@
 
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
-
+  <script>
+      var global = global || window;
```
Comment (reinoutbossuyt) [inline src/index.html]: Sure we need this? Can't find a global.varname in the app, Buffer and process also.

File: src/app/core/user/services/user.service.ts [hunk 8]
```diff
@@ -33,13 +34,13 @@ export class UserService {
...
+    return this.getUser().pipe(map(() => {
       return true;
     }, () => {
       return false;
-    }).catch(error => {
-      return Observable.of(false);
-    });
+    }),catchError(error => {
+      return observableOf(false);
+    }),);
```
[the remark this answers is not in the document]
Comment (brampauwelyn) [reply src/app/core/user/services/user.service.ts]: fixed

File: src/index.html (hunk 7 above)
Comment (brampauwelyn) [reply src/index.html]: Found a more elegant solution added this fix in the polyfills.ts

**Pull request merged**